Fix openlibm build

dvdfreitag opened the pull request:
The openlibm build was broken because of a makefile issue. `openlibm_OBJECT_FILES` wasn't being defined, so the build was creating an empty library.

- Upgraded openlibm build system and integrated with the new system. `openlibm_SRCS` and `openlibm_OBJECT_FILES` redefined in `mk/openlibm.mk` using a wildcard.
- Removed unnecessary sources in openlibm (there were some files which aren't needed for aarch64)
- Fixed a few include issues with openlibm itself
- Added an entry to the gitignore so that the xz Automake m4 cache is ignored

dvdfreitag committed:
Ignore xz Automake cache

Automake creates a directory called autom4te which contains cached build outputs. It should not be checked in.

dvdfreitag committed:
Fix openlibm build

Remove unneeded files and fix openlibm.mk so that the correct source files are found, build, and linked into build/openlibm/libm.a

roblabla commented:
Sorry for the wait :(.

https://github.com/dvdfreitag/libtransistor/blob/cb503770aae2edb84ed4e6319c9873396f0efe1b/openlibm/include/openlibm.h still references `openlibm_math.h` and `openlibm_complex.h`, I suppose those should be changed to `math.h` and `complex.h` respectively ?

Otherwise, this all looks good. I checked locally that it compiles and works. It does \o/.

dvdfreitag commented:
Ah, yeah. That file isn't actually included anywhere. It's there so you can include all of openlibm without poisoning your include paths with an additional `math.h` and `complex.h`.

dvdfreitag committed:
Delete openlibm.h

openlibm.h is unnecessary for our purposes, it's a holdover from the old include system. It exists so that the include paths aren't poisoned with extra `math.h` or `complex.h`
